modules/core/src/v2/coins/sol.ts

```diff
@@ -133,15 +157,32 @@ export class Sol extends BaseCoin {
     params: SolSignTransactionOptions,
     callback?: NodeCallback<SignedTransaction>
   ): Bluebird<SignedTransaction> {
-    throw new MethodNotImplementedError('signTransaction method not implemented');
+    const self = this;
+    return co<SignedTransaction>(function* () {
+      const factory = accountLib.register(self.getChain(), accountLib.Sol.TransactionBuilderFactory);
+      const txBuilder = factory.from(params.txPrebuild.txBase64);
+      txBuilder.sign({ key: params.prv });
+      const transaction: any = yield txBuilder.build();
+
+      if (!transaction) {
+        throw new Error('Invalid transaction');
+      }
+
+      const response = {
+        txBase64: transaction.toBroadcastFormat(),
+      };
+      return transaction.signature.length >= 2 ? response : { halfSigned: response };
```

Comment:
@rliberal-simtlix @joey-bai 
Should we just return the response without this check? Since there is no multi sign for Solana, at least for most operations that are not secondary wallet creation

Reply:
to start, lets assume this is only for single sig.
we will need to update this impl when TSS is ready.